src/main/java/org/springframework/data/couchbase/core/ReactiveFindByQueryOperationSupport.java

```diff
@@ -124,6 +133,15 @@ private String assembleEntityQuery(final boolean count) {
 			return query.toN1qlString(template, this.domainType, count);
 		}
 
+		@Override
+		public FindWithProjection<T> inCollection(String collection) {
```

Comment:
`inCollection`/`distinct` methods are MongoDB-specific features. It makes only sense to re-apply the same pattern if Couchbase can provide such functionality.